crates/oxc_linter/src/rules/deepscan/uninvoked_array_callback.rs

```diff
@@ -0,0 +1,96 @@
+use oxc_ast::{
+    ast::{Argument, Expression, MemberExpression},
+    AstKind, GetSpan, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Uninvoked array callback")]
+#[diagnostic(
+    severity(warning),
+    help("consider filling the array with `undefined` values using `Array.prototype.fill()`")
+)]
+struct UninvokedArrayCallbackDiagnostic(
+    #[label("this callback function will not be invoked")] Span,
+    #[label("because this is an array with only empty slots")] Span,
+);
+
+/// `https://deepscan.io/docs/rules/uninvoked-array-callback`
+#[derive(Debug, Default, Clone)]
+pub struct UninvokedArrayCallback;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule applies when the Array functions having callback argument are used for an array with empty slots.
+    ///
+    /// ### Why is this bad?
+    /// When the Array constructor is called with a single number argument, an array with the specified number of empty slots (not actual undefined values) is constructed.
```

Comment:
This is basically what we'd consider a sparse array right? For example, `Array(10).length` returns 10 but it doesn't actually have any inner values

Reply:
Yes, I just copied the deepscan documentation.